File: src/1165-Single-Row-Keyboard/1165.py
class Solution:
    def calculateTime(self, keyboard: str, word: str) -> int:
        d = dict()
        for i, v in enumerate(keyboard):
            d[v] = i
        
        res, pre = 0, 0
        for c in word:
            res += abs(d[c] - pre)
            pre = d[c]
        return res

File: src/1165-Single-Row-Keyboard/1165.cpp
class Solution 
{
public:
    int calculateTime(string keyboard, string word) 
    {
        unordered_map<char, int> d;
        for (int i = 0; i < keyboard.size(); ++i)
        {
            d[keyboard[i]] = i;
        }
        
        int res = 0, pre = 0;
        for (char c : word)
        {
            res += abs(d[c] - pre);
            pre = d[c];
        }
        return res;
    }
};